tests/plugins/test_check_and_set_platforms.py

```diff
@@ -341,3 +338,63 @@ def test_platforms_from_cluster_config(tmpdir, user_params,
         assert plugin_result[PLUGIN_CHECK_AND_SET_PLATFORMS_KEY] == set(result)
     else:
         assert plugin_result[PLUGIN_CHECK_AND_SET_PLATFORMS_KEY] is None
+
+
+@pytest.mark.parametrize(('koji_platforms', 'cluster_platforms', 'result', 'skips', 'fails'), [
+    (None, None, None, None, None),
+    (['x86_64'], None, None, None, 'no_platforms'),
+    (['x86_64'], {'ppc64le': True}, None, None, 'no_platforms'),
+    (['x86_64', 'ppc64le'], {'x86_64': True, 'ppc64le': True}, ['x86_64', 'ppc64le'], None, None),
+    (['x86_64', 'ppc64le'], {'x86_64': False, 'ppc64le': True}, None, None, 'disabled'),
+    (['x86_64', 'ppc64le'], {'x86_64': False, 'ppc64le': False}, None, None, 'disabled'),
+    (['x86_64', 'ppc64le'], {'x86_64': True}, ['x86_64'], ['ppc64le'], None),
+    (['x86_64', 'ppc64le', 's390x'], {'x86_64': True}, ['x86_64'], ['ppc64le', 's390x'], None),
+])
+def test_disabled_clusters(tmpdir, caplog, user_params, koji_platforms,
+                           cluster_platforms, result, skips, fails):
+    write_container_yaml(tmpdir)
+
+    tasker, workflow = prepare(tmpdir)
+
+    build_json = {'metadata': {'labels': {}}}
+    flexmock(util).should_receive('get_build_json').and_return(build_json)
+
+    new_koji_platforms = None
+    if koji_platforms:
+        new_koji_platforms = {k: True for k in koji_platforms}
+    session = mock_session(new_koji_platforms)
+    flexmock(koji_util).should_receive('create_koji_session').and_return(session)
+    set_reactor_config_map(workflow, cluster_platforms)
+
+    runner = PreBuildPluginsRunner(tasker, workflow, [{
+        'name': PLUGIN_CHECK_AND_SET_PLATFORMS_KEY,
+        'args': {'koji_target': KOJI_TARGET},
+    }])
+
+    if fails:
+        with pytest.raises(Exception) as e:
+            runner.run()
+
+        if fails == 'no_platforms':
+            msg = 'No platforms to build for'
+        elif fails == 'disabled':
+            msg = 'Platforms specified in config map, but have all clusters disabled'
```

Comment:
Minor thing, the error message could be incorporated directly into the parameters, and then it would not have to be assigned in the code based on fails parameter. and we would have only fails_with_error_msg or something like that.

Reply:
yeah I know, but that message is quite long, so I didn't want to add it to parameters

Reply:
I feel like long message is better than additional conditionals but it's not important.